packages/calcite-components/src/tests/commonTests.ts

```diff
@@ -983,6 +1032,80 @@ export function formAssociated(
 
     expect(called).toBe(true);
   }
+
+  async function assertPreventsFormSubmission(
+    page: E2EPage,
+    component: E2EElement,
+    submitButton: E2EElement,
+    message: string,
+  ) {
+    await submitButton.click();
+    await page.waitForChanges();
+
+    await expectValidationInvalid(component, message);
+  }
+
+  async function assertClearsValidationOnValueChange(
+    page: E2EPage,
+    component: E2EElement,
+    options: FormAssociatedOptions,
+    event: EventSpy,
+    tag: string,
+  ) {
+    await component.callMethod("setFocus");
+    await page.waitForChanges();
+
+    if (tag === "calcite-combobox") {
```

Comment:
Is there a way to move this component specific logic outside of the common test? Maybe this kind of data could be stored in the component resource file or something and passed in as an option on the test?

Reply:
Just the combobox, select and input-time picker having specific logic here seems odd since its a common test.